lessons/advanced/gen-stage-1.md

```diff
@@ -0,0 +1,368 @@
+---
```

Comment:
I was thinking of adding the `_1` because I'm working on another lesson that builds this. Wasnt sure what would be best because we dont really have any examples that build upon each other that are complex-ish like this that I know of